lib/utils/interval/interval.go

```diff
@@ -129,6 +140,15 @@ func (i *Interval) run(timer *time.Timer) {
 			// output channel is set.
 			timer.Reset(i.duration())
 			ch = i.ch
+		case <-i.reset:
+			// stop and drain timer
+			if !timer.Stop() {
+				<-timer.C
+			}
```

Comment:
My brain is kind of fried after reading through all of this. I can't say for sure whether this is correct or not, but looks like a particularly tricky spot. Are you confident we got this right here? ( timer stopped twice, drain necessary, etc)

Reply:
This method of stop/drain is lifted verbatim from the official docs for how to handle stopping a timer in a concurrent context where you aren't certain if it has fired and/or been drained.